lib/ex_doc/cli.ex

```diff
@@ -23,7 +23,6 @@ defmodule ExDoc.CLI do
         ],
         switches: [
           language: :string,
-          proglang: :atom,
```

Comment:
Shouldn't we still be able to pass it as an argument from the CLI? 🤔

Reply:
yup, I think I made this change by mistake, will revert.

it brings an interesting question though, what the proglang is actually used for, and perhaps it is a bit overloaded already. I could even group the contract callbacks into 3 groups:

- `@callback id()` - this is used for the "theme", picking the correct CSS, etc. This is a **project-wide** setting
- `@callback filter_prefix_pattern()` - `"Elixir.#{filter_prefix}*.beam"` for Elixir, `"#{filter_prefix}*.beam"` for Erlang. This is a **project-wide** setting
- remaining callbacks are **per-module**

so in a way, the `--proglang` flag only matters for the first two callbacks.

I think it is fine to move forward with things as is but just adding some more context around this.

Reply:
That's a good point, because it raises a question about mixed projects and other configuration, like group nesting. For now, I think we need to at least change how filtering works:

1. First get all modules from disk
2. And then, based on the language, filter the module

Effectively moving it from a project config to a module config. WDYT?

Reply:
sounds good, thanks. I'll leave the `filter_prefix_pattern` callback as is, I'll just change how it is being used internally.

Reply:
I removed the `filter_prefix_pattern/1` callback in 49d3fcd0fa50a15c51ad4f2ef63bae6b0d580d4c

in fact, I removed the `id/0` callback too in b4d1fcfd545d670f5b684dcc5d4e6a38a7395f1c.

Now, `proglang` is the project-wide option for CSS and such, and `language` (as in ExDoc.Language) is per-module.

Thanks for the feedback! <3